src/javascripts/modules/pillbar.js

```diff
@@ -1,53 +1,33 @@
 export default class Pillbar {
   constructor(el) {
     this.el = el
-    this.setVars()
-    this.setupListeners()
+    this.createObserver()
   }
 
-  setVars(el) {
-    /* Ticking essentially == isRequestingAnimationFrame; I'm leaving
-    it as 'ticking' to follow requestingAnimationFrame convention/docs */
-    this.ticking = false
-    this.lastScrollY = 0
-    this.el.pillbarOffset = this.el.offsetTop
-  }
-
-  setupListeners() {
-    window.addEventListener('resize', () => this.handleResize())
-    window.addEventListener('scroll', () => this.handleScroll())
-  }
+  createObserver() {
+    this.observer
 
-  handleScroll() {
-    this.lastScrollY = window.scrollY
-    if (!this.ticking) {
-      window.requestAnimationFrame(() => {
-        this.togglePillbar()
-      })
+    this.options = {
+      root: null,
+      rootMargin: "0px",
+      threshold: 0,
     }
-    this.ticking = true
-  }
 
-  handleResize() {
-    clearTimeout(this.timeout)
-    this.timeout = setTimeout(() => this.getPillbarOffset(), 200)
+    this.observer = new IntersectionObserver(this.handleIntersect, this.options)
+    this.observer.observe(document.querySelector('.title-box'))
   }
 
-  getPillbarOffset() {
-    if(!this.el.isPinned){
-      this.el.pillbarOffset = this.el.offsetTop
-    }
-  }
-
-  togglePillbar() {
-    this.ticking = false
-
-    if (this.lastScrollY > this.el.pillbarOffset) {
-      this.el.classList.add('pillbar-fixed')
-      this.el.isPinned = true
-    } else {
-      this.el.classList.remove('pillbar-fixed')
-      this.el.isPinned = false
-    }
+  handleIntersect(entries, observer) {
+    entries.forEach(function(entry) {
+      if (entry.intersectionRatio > 0 && !entry.isRequestingAnimationFrame) {
```

Comment:
Where does `isRequestingAnimationFrame` come from here?

Reply:
Turns out it wasn't coming from anywhere. I just pushed up a fix to that.